pkg/utils/plugin/plugin.go

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+package plugin
+
+import (
+	"sync"
+
+	"github.com/azure/kaito/pkg/inference"
+)
+
+type Registration struct {
+	Name     string
+	Instance inference.Model
+}
+
+type ModelRegister struct {
+	sync.RWMutex
```

Comment:
sync.Map?
https://stackoverflow.com/questions/61355008/is-sync-map-the-right-tool-for-this-use-case-or-is-a-rw-mutex-lock-better

Reply:
They are interchangeable. With sync.Map, you need to call Load/Store() and you cannot easily implement MustGet.